storage/persistence/spanner/spanner.go

```diff
@@ -163,6 +163,35 @@ func (s *Spanner) GetReadyJobs(ctx context.Context, req *model.JobDataReq) (jobs
 	return jobs, nil
 }
 
+// BatchGetJobsByID returns job data by job ID
+func (s *Spanner) BatchGetJobsByID(ctx context.Context, IDs []string) (jobs []*model.JobData, err error) {
+	txn := s.cli.ReadOnlyTransaction()
+	defer txn.Close()
+
+	for _, id := range IDs {
+		iter := txn.Query(ctx, spanner.Statement{
+			SQL: "SELECT pool_name, job_id, namespace, queue, body, ready_time, expired_time, created_time, tries " +
+				"FROM lmstfy_jobs WHERE job_id=@jobid LIMIT @limit",
```

Comment:
```suggestion
				"FROM lmstfy_jobs WHERE job_id in IN UNNEST(@ids)",
```

Can use `IN` condition to query jobs at once